### PR TITLE
Update/swift code style

### PR DESCRIPTION
**Pull request by zenangst**
- [x] Update `Use of Self` section
- [x] Add section about `forEach`
- [x] Add sentence about when to use `guard`

### CODE REVIEW

**Comment by zenangst** [comment]
@hyperoslo/ios What do you guys think about this?

File: style-guidelines/Swift.md [hunk 1]
```diff
@@ -169,6 +169,12 @@ if let textContainer = self.textContainer {
 **Not Preferred:**
 
 ```swift
+if let textContainer = self.textContainer {
+  // do many things with textContainer
+}
+```
+
+```swift
 if let maybeThisCouldBeTextContainer = textContainer {
```
Comment (RamonGilabert) [inline style-guidelines/Swift.md]: Know you didn't change it here, but that's bad.

**Comment by 3lvis** [comment]
https://github.com/apple/swift-evolution/blob/master/proposals/0009-require-self-for-accessing-instance-members.md

File: style-guidelines/Swift.md [hunk 2]
```diff
@@ -460,6 +477,43 @@ var faxNumber: Optional<Int>
...
+
+Prefer `forEach` over `for-in` when applicable.
+
+**Preferred:**
+
+Use named parameters when the object is being referenced more than once.
+
+```swift
+attendeeList.forEach { attendee in 
+print("\(attendee.name) is attending with \(attendee.guests.count) guests.")
```
Comment (vadymmarkov) [inline style-guidelines/Swift.md]: I'd put an indent.

File: style-guidelines/Swift.md [hunk 3]
```diff
@@ -370,11 +381,17 @@ textContainer?.textLabel?.setNeedsDisplay()
 Use optional binding when it's more convenient to unwrap once and perform multiple operations:
 
 ```swift
-if let textContainer = self.textContainer {
+if let textContainer = textContainer {
   // do many things with textContainer
 }
 ```
 
+Use guard unwrapping if the object is required for continuing the operation.
```
Comment (RamonGilabert) [inline style-guidelines/Swift.md]: We could argue that if you are for instance in the middle of the method and you only want to use it for one thing, though.

**Comment by zenangst** [comment]
@3lvis saw that, still under review. We would obviously change it to make it work if the proposal is in fact approved. #kthnxbye

File: style-guidelines/Swift.md (hunk 1 above)
Comment (zenangst) [inline style-guidelines/Swift.md]: Yeah but it is in the **Not Preferred:** section :sunglasses:

**Comment by zenangst** [comment]
@hyperoslo/ios we good meow? :grin:

**Comment by vadymmarkov** [comment]
@zenangst :100:

**Pull request merged**